cfme/tests/services/test_ansible_workflow_servicecatalogs.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-:
+import pytest
+
+from cfme import test_requirements
+from cfme.services.myservice import MyService
+from cfme.services.service_catalogs import ServiceCatalogs
+from cfme.utils import testgen
+from cfme.utils.blockers import GH
+from cfme.utils.log import logger
+
+
+pytestmark = [
+    test_requirements.service,
+    pytest.mark.tier(2),
+    pytest.mark.ignore_stream('upstream')]
+
+
+def pytest_generate_tests(metafunc):
+    # Filter out providers without provisioning data or hosts defined
+    argnames, argvalues, idlist = testgen.config_managers(metafunc)
+    new_idlist = []
+    new_argvalues = []
+    for i, argvalue_tuple in enumerate(argvalues):
```

Comment:
This `for` loop could be shortened with list comprehension:

```python,
list_of_tuples = [ 
    (i, argval) for i, argval in enumerate(argvalues) if dict(zip(argnames, argval))["config_manager_obj"].yaml_data["provisioning"] 
]
new_idlist = [x[0] for x in list_of_tuples]
new_argvalues = [x[1] for x in list_of_tuples]
```

Ha, after typing that all out, I'm not convinced it's any better, I'll leave it as optional if you want to refactor.